src/ng/directive/select.js

```diff
@@ -607,8 +607,24 @@ var selectDirective = function() {
         // Write value now needs to set the selected property of each matching option
         selectCtrl.writeValue = function writeMultipleValue(value) {
           forEach(element.find('option'), function(option) {
-            option.selected = !!value && (includes(value, option.value) ||
-                                          includes(value, selectCtrl.selectValueMap[option.value]));
+            var shouldBeSelected = !!value && (includes(value, option.value) ||
+                                               includes(value, selectCtrl.selectValueMap[option.value]));
+            var currentlySelected = option.selected;
+
+            // IE and Edge, adding options to the selection via shift+click/UP/DOWN,
+            // will de-select already selected options if "selected" on those options was set
+            // more than once (i.e. when the options were already selected)
+            // So we only modify the selected property if neccessary.
+            // Note: this behavior cannot be replicated via unit tests because it only shows in the
+            // actual user interface.
+            if ((shouldBeSelected && !currentlySelected)) {
+              setOptionAsSelected(jqLite(option));
+            } else if ((!shouldBeSelected && currentlySelected)) {
+              option.selected  = shouldBeSelected;
```

Comment:
I wonder why we are using `option.selected` here, but `option.prop('selected', true)` in `setOptionAsSelected()`.
Probably doesn't matter, so we should use the same for consistency. In fact, I think it is better to make `setOptionSelected()` a helper that can both select and unselect an option and use that whenever needed.